charts/consul/test/terraform/eks/main.tf

```diff
@@ -84,4 +87,86 @@ data "aws_eks_cluster" "cluster" {
 data "aws_eks_cluster_auth" "cluster" {
   count = var.cluster_count
   name  = module.eks[count.index].cluster_id
+}
+
+# The following resources are meant to apply when cluster_count=2 to set up vpc peering and the appropriate routes and
+# security groups so traffic between vpc's is allowed. Since the pipeline only ever deploys with a cluster_count of 2,
+# these resources are hardcoded to work in an environment with 2 clusters.
+
+# Each EKS cluster needs to allow ingress traffic from the other VPC.
+resource "aws_security_group_rule" "allowingressfrom1-0" {
+  type              = "ingress"
+  from_port         = 0
+  to_port           = 65535
+  protocol          = "tcp"
+  cidr_blocks       = [module.vpc[1].vpc_cidr_block]
+  security_group_id = module.eks[0].cluster_primary_security_group_id
+}
+
+resource "aws_security_group_rule" "allowingressfrom0-1" {
+  type              = "ingress"
+  from_port         = 0
+  to_port           = 65535
+  protocol          = "tcp"
+  cidr_blocks       = [module.vpc[0].vpc_cidr_block]
+  security_group_id = module.eks[1].cluster_primary_security_group_id
+}
+
+# Create a peering connection. This is the requester's side of the connection.
+resource "aws_vpc_peering_connection" "peer" {
+  vpc_id        = module.vpc[0].vpc_id
+  peer_vpc_id   = module.vpc[1].vpc_id
+  peer_owner_id = data.aws_caller_identity.caller.account_id
+  peer_region   = var.region
+  auto_accept   = false
+
+  tags = {
+    Side = "Requester"
+  }
+}
+
+# Accepter's side of the vpc peering connection.
+resource "aws_vpc_peering_connection_accepter" "peer" {
+  vpc_peering_connection_id = aws_vpc_peering_connection.peer.id
+  auto_accept               = true
+
+  tags = {
+    Side = "Accepter"
+  }
+}
+
+# Get the route table ids that are associated with eks cluster 0, so we can create routes in those route tables.
+data "aws_route_tables" "rtseks0" {
+  vpc_id = module.vpc[0].vpc_id
+
+  filter {
+    name   = "tag:Name"
+    values = [format("%s-*", module.eks[0].cluster_id)]
+  }
+}
```

Comment:
I'm curious why do we need a data resource if we have the vpc module created in this terraform. In my experience, I usually use data resource for resources that exist in a different terraform state.

Should this be a local variable instead?

Reply:
I think you can use vpc module output so you can combine public and private route table ids into one array: `[module.vpc[count.index].public_route_table_ids[0], module.vpc[count.index].private_route_table_ids[0]]`

Reply:
I think the vpc module creates two route tables that are shared between subnets. So there should be one private and one public route table per VPC

Reply:
sorry for missing this earlier! your suggestion worked for me and I was able to remove the route table data block!